.github/workflows/library_interop_tests.yml

```diff
@@ -0,0 +1,189 @@
+        language: [java, net]
+        # https://taskei.amazon.dev/tasks/CrypTool-5284
+        dotnet-version: ["6.0.x"]
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+
+      - uses: actions/checkout@v3
+      #  Not all submodules are needed.
+      # We manually pull the submodule we DO need.
+      - run: git submodule update --init libraries
+      - run: git submodule update --init smithy-dafny
+
+      # Set up runtimes
+      - name: Setup .NET Core SDK ${{ matrix.dotnet-version }}
+        if: matrix.language == 'net'
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+
+      - name: Setup Java 17
+        if: matrix.language == 'java'
+        uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: 17
+
+      - name: Setup Dafny
+        uses: dafny-lang/setup-dafny-action@v1.6.1
+        with:
+          dafny-version: ${{ inputs.dafny }}
+
+      # Build implementation for each runtime
+      - name: Build ${{ matrix.library }} implementation in Java
+        if: matrix.language == 'java'
+        shell: bash
+        working-directory: ./${{ matrix.library }}
+        run: |
+          # This works because `node` is installed by default on GHA runners
+          CORES=$(node -e 'console.log(os.cpus().length)')
+          make build_java CORES=$CORES
+
+      - name: Build ${{ matrix.library }} implementation in .NET
+        if: matrix.language == 'net'
+        shell: bash
+        working-directory: ./${{ matrix.library }}
+        run: |
+          # This works because `node` is installed by default on GHA runners
+          CORES=$(node -e 'console.log(os.cpus().length)')
+          make transpile_net
+
+      - name: Setup gradle
+        if: matrix.language == 'java'
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.2
+
+      # Test Vectors need to call KMS
+      - name: Configure AWS Credentials for Tests
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::370957321024:role/GitHub-CI-MPL-Dafny-Role-us-west-2
+          role-session-name: InterOpTests
+
+      - name: Create Manifests
+        working-directory: ./${{ matrix.library }}
+        run: make test_generate_vectors_${{ matrix.language }}
+
+      - name: Create Encrypt Manifests
+        working-directory: ./${{ matrix.library }}
+        run: make test_encrypt_vectors_${{ matrix.language }}
+
+      - name: Upload Encrypt Manifest and keys.json files
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{matrix.os}}_vector_artifact_${{matrix.language}}
+          path: ./${{matrix.library}}/runtimes/${{matrix.language}}/*.json
```

Comment:
Why not upload the binary as well?
Then you don't need to build it again on the other side.

Reply:
I can see how this might make the make targets more complicated.
But it is a though :)

Reply:
That would be nice to have, and we will definitely have to have it when we publish the TestVectors. However; for now this gets us to easy interop testing in CI (coats of paint)